store/cache/cache.go

```diff
@@ -91,7 +91,7 @@ func (cmgr *CommitKVStoreCacheManager) Reset() {
 
 // CacheWrap implements the CacheWrapper interface
 func (ckv *CommitKVStoreCache) CacheWrap() types.CacheWrap {
-	return cachekv.NewStore(ckv)
+	return cachekv.NewStore(ckv, types.NewInfiniteGasMeter(), types.KVGasConfig())
```

Comment:
I'm not sure if using `NewInfiniteGasMeter` here is correct, please let me know if it should be something else

Reply:
I don't think it's correct. It should be the same GasMeter that's in `sdk.Context`.

Reply:
I think we use `InfiniteGasMeter` in `sdk.Context`. See [here](https://github.com/cosmos/cosmos-sdk/blob/main/types/context.go#L108)

Reply:
Yeah, but the antehandlers replaces it with a GasMeter that's limited to the tx's gas limit. We should consume gas on that GasMeter.